crates/uv/tests/it/python_find.rs

```diff
@@ -707,3 +707,190 @@ fn python_required_python_major_minor() {
     error: No interpreter found for Python >3.11.[X], <3.12 in virtual environments, managed installations, or search path
     "###);
 }
+
+#[test]
+fn python_find_script() {
+    let context = TestContext::new("3.13");
+    let filters = context
+        .filters()
+        .into_iter()
+        .chain(vec![(
+            r"environments-v2/[\w-]+",
+            "environments-v2/[HASHEDNAME]",
+        )])
+        .collect::<Vec<_>>();
+
+    uv_snapshot!(filters, context.init().arg("--script").arg("foo.py"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Initialized script at `foo.py`
+    "###);
+
+    uv_snapshot!(filters, context.sync().arg("--script").arg("foo.py"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Creating script environment at: [CACHE_DIR]/environments-v2/[HASHEDNAME]
+    Resolved in [TIME]
+    Audited in [TIME]
+    "###);
+
+    if cfg!(windows) {
+        uv_snapshot!(filters, context.python_find().arg("--script").arg("foo.py"), @r###"
+        success: true
+        exit_code: 0
+        ----- stdout -----
+        [CACHE_DIR]/environments-v2/[HASHEDNAME]/Scripts/python.exe
```

Comment:
There's `TestContext::with_filtered_exe_suffix`, `with_filtered_virtualenv_bin`, and `with_filtered_python_names` for this.